chore: upgrade to Vue 3.5

makedopamine opened the pull request:
Note to reviewers: It might be easier to review this commit by commit.

makedopamine committed:
chore: upgrade deps

makedopamine committed:
chore: replace __ExtractPublicPropTypes with ExtractPublicPropTypes

makedopamine committed:
fix: get rid of type errors

makedopamine committed:
fix: resolve test errors with @vue/test-utils v2.4.6

makedopamine committed:
fix: resolve test errors with Vue 3.5.22

Dsaquel commented on package.json:
```diff
@@ -59,7 +59,7 @@
     "@types/lodash": "^4.14.182",
     "@types/lodash-es": "^4.17.6",
-    "@vueuse/core": "^10.11.0",
+    "@vueuse/core": "catalog:",
```
When i try downloading this version of EP in a project i have this error: 

```sh
pnpm create vue
...
pnpm add -D https://pkg.pr.new/element-plus/element-plus@22096
```

***Error:***
ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  @vueuse/core@catalog: isn't supported by any available resolver.

<details>
<summary>full message error</summary>
 ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  @vueuse/core@catalog: isn't supported by any available resolver.

This error happened while installing the dependencies of element-plus@2.11.3

An external package outside of the pnpm workspace declared a dependency using
the catalog protocol. This is likely a bug in that external package. Only
packages within the pnpm workspace may use catalogs. Usages of the catalog
protocol are replaced with real specifiers on 'pnpm publish'.

This is likely a bug in the publishing automation of this package. Consider filing
a bug with the authors of:

  element-plus@2.11.3
</details>

Dsaquel replied:
Using npm also have an error but there are no explicit error, surely the error above using pnpm is the same as npm.

Dsaquel committed:
ci: set pnpm flag

makedopamine committed:
chore: update the Vue peer dependency version

warmthsea reviewed (commented):
The example code here needs to be changed.

https://github.com/element-plus/element-plus/blob/dopamine/upgrade_vue/docs/examples/tree/customized-node.vue#L86-L104

 `'Append '` -> `{ default: () => 'Append ' }`
`'Delete'`     -> `{ default: () => 'Delete' }`

[Now](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiY3VzdG9tLXRyZWUtY29udGFpbmVyXCI+XG4gICAgPHA+VXNpbmcgcmVuZGVyLWNvbnRlbnQ8L3A+XG4gICAgPGVsLXRyZWVcbiAgICAgIHN0eWxlPVwibWF4LXdpZHRoOiA2MDBweFwiXG4gICAgICA6ZGF0YT1cImRhdGFTb3VyY2VcIlxuICAgICAgc2hvdy1jaGVja2JveFxuICAgICAgbm9kZS1rZXk9XCJpZFwiXG4gICAgICBkZWZhdWx0LWV4cGFuZC1hbGxcbiAgICAgIDpleHBhbmQtb24tY2xpY2stbm9kZT1cImZhbHNlXCJcbiAgICAgIDpyZW5kZXItY29udGVudD1cInJlbmRlckNvbnRlbnRcIlxuICAgIC8+XG4gICAgPHA+VXNpbmcgc2NvcGVkIHNsb3Q8L3A+XG4gICAgPGVsLXRyZWVcbiAgICAgIHN0eWxlPVwibWF4LXdpZHRoOiA2MDBweFwiXG4gICAgICA6ZGF0YT1cImRhdGFTb3VyY2VcIlxuICAgICAgc2hvdy1jaGVja2JveFxuICAgICAgbm9kZS1rZXk9XCJpZFwiXG4gICAgICBkZWZhdWx0LWV4cGFuZC1hbGxcbiAgICAgIDpleHBhbmQtb24tY2xpY2stbm9kZT1cImZhbHNlXCJcbiAgICA+XG4gICAgICA8dGVtcGxhdGUgI2RlZmF1bHQ9XCJ7IG5vZGUsIGRhdGEgfVwiPlxuICAgICAgICA8ZGl2IGNsYXNzPVwiY3VzdG9tLXRyZWUtbm9kZVwiPlxuICAgICAgICAgIDxzcGFuPnt7IG5vZGUubGFiZWwgfX08L3NwYW4+XG4gICAgICAgICAgPGRpdj5cbiAgICAgICAgICAgIDxlbC1idXR0b24gdHlwZT1cInByaW1hcnlcIiBsaW5rIEBjbGljaz1cImFwcGVuZChkYXRhKVwiPlxuICAgICAgICAgICAgICBBcHBlbmRcbiAgICAgICAgICAgIDwvZWwtYnV0dG9uPlxuICAgICAgICAgICAgPGVsLWJ1dHRvblxuICAgICAgICAgICAgICBzdHlsZT1cIm1hcmdpbi1sZWZ0OiA0cHhcIlxuICAgICAgICAgICAgICB0eXBlPVwiZGFuZ2VyXCJcbiAgICAgICAgICAgICAgbGlua1xuICAgICAgICAgICAgICBAY2xpY2s9XCJyZW1vdmUobm9kZSwgZGF0YSlcIlxuICAgICAgICAgICAgPlxuICAgICAgICAgICAgICBEZWxldGVcbiAgICAgICAgICAgIDwvZWwtYnV0dG9uPlxuICAgICAgICAgIDwvZGl2PlxuICAgICAgICA8L2Rpdj5cbiAgICAgIDwvdGVtcGxhdGU+XG4gICAgPC9lbC10cmVlPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBFbEJ1dHRvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuaW1wb3J0IHR5cGUgeyBSZW5kZXJDb250ZW50Q29udGV4dCwgUmVuZGVyQ29udGVudEZ1bmN0aW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5pbnRlcmZhY2UgVHJlZSB7XG4gIGlkOiBudW1iZXJcbiAgbGFiZWw6IHN0cmluZ1xuICBjaGlsZHJlbj86IFRyZWVbXVxufVxudHlwZSBOb2RlID0gUmVuZGVyQ29udGVudENvbnRleHRbJ25vZGUnXVxudHlwZSBEYXRhID0gUmVuZGVyQ29udGVudENvbnRleHRbJ2RhdGEnXVxuXG5sZXQgaWQgPSAxMDAwXG5cbmNvbnN0IGFwcGVuZCA9IChkYXRhOiBEYXRhKSA9PiB7XG4gIGNvbnN0IG5ld0NoaWxkID0geyBpZDogaWQrKywgbGFiZWw6ICd0ZXN0dGVzdCcsIGNoaWxkcmVuOiBbXSB9XG4gIGlmICghZGF0YS5jaGlsZHJlbikge1xuICAgIGRhdGEuY2hpbGRyZW4gPSBbXVxuICB9XG4gIGRhdGEuY2hpbGRyZW4ucHVzaChuZXdDaGlsZClcbiAgZGF0YVNvdXJjZS52YWx1ZSA9IFsuLi5kYXRhU291cmNlLnZhbHVlXVxufVxuXG5jb25zdCByZW1vdmUgPSAobm9kZTogTm9kZSwgZGF0YTogRGF0YSkgPT4ge1xuICBjb25zdCBwYXJlbnQgPSBub2RlLnBhcmVudFxuICBjb25zdCBjaGlsZHJlbjogVHJlZVtdID0gcGFyZW50Py5kYXRhLmNoaWxkcmVuIHx8IHBhcmVudD8uZGF0YVxuICBjb25zdCBpbmRleCA9IGNoaWxkcmVuLmZpbmRJbmRleCgoZCkgPT4gZC5pZCA9PT0gZGF0YS5pZClcbiAgY2hpbGRyZW4uc3BsaWNlKGluZGV4LCAxKVxuICBkYXRhU291cmNlLnZhbHVlID0gWy4uLmRhdGFTb3VyY2UudmFsdWVdXG59XG5cbmNvbnN0IHJlbmRlckNvbnRlbnQ6IFJlbmRlckNvbnRlbnRGdW5jdGlvbiA9IChoLCB7IG5vZGUsIGRhdGEgfSkgPT4ge1xuICByZXR1cm4gaChcbiAgICAnZGl2JyxcbiAgICB7XG4gICAgICBjbGFzczogJ2N1c3RvbS10cmVlLW5vZGUnLFxuICAgIH0sXG4gICAgW1xuICAgICAgaCgnc3BhbicsIG51bGwsIG5vZGUubGFiZWwpLFxuICAgICAgaCgnZGl2JywgbnVsbCwgW1xuICAgICAgICBoKFxuICAgICAgICAgIEVsQnV0dG9uLFxuICAgICAgICAgIHtcbiAgICAgICAgICAgIHR5cGU6ICdwcmltYXJ5JyxcbiAgICAgICAgICAgIGxpbms6IHRydWUsXG4gICAgICAgICAgICBvbkNsaWNrOiAoKSA9PiBhcHBlbmQoZGF0YSksXG4gICAgICAgICAgfSxcbiAgICAgICAgICAnQXBwZW5kICdcbiAgICAgICAgKSxcbiAgICAgICAgaChcbiAgICAgICAgICBFbEJ1dHRvbixcbiAgICAgICAgICB7XG4gICAgICAgICAgICB0eXBlOiAnZGFuZ2VyJyxcbiAgICAgICAgICAgIGxpbms6IHRydWUsXG4gICAgICAgICAgICBzdHlsZTogJ21hcmdpbi1sZWZ0OiA0cHgnLFxuICAgICAgICAgICAgb25DbGljazogKCkgPT4gcmVtb3ZlKG5vZGUsIGRhdGEpLFxuICAgICAgICAgIH0sXG4gICAgICAgICAgJ0RlbGV0ZSdcbiAgICAgICAgKSxcbiAgICAgIF0pLFxuICAgIF1cbiAgKVxufVxuXG5jb25zdCBkYXRhU291cmNlID0gcmVmPFRyZWVbXT4oW1xuICB7XG4gICAgaWQ6IDEsXG4gICAgbGFiZWw6ICdMZXZlbCBvbmUgMScsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgaWQ6IDQsXG4gICAgICAgIGxhYmVsOiAnTGV2ZWwgdHdvIDEtMScsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAge1xuICAgICAgICAgICAgaWQ6IDksXG4gICAgICAgICAgICBsYWJlbDogJ0xldmVsIHRocmVlIDEtMS0xJyxcbiAgICAgICAgICB9LFxuICAgICAgICAgIHtcbiAgICAgICAgICAgIGlkOiAxMCxcbiAgICAgICAgICAgIGxhYmVsOiAnTGV2ZWwgdGhyZWUgMS0xLTInLFxuICAgICAgICAgIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG4gIHtcbiAgICBpZDogMixcbiAgICBsYWJlbDogJ0xldmVsIG9uZSAyJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBpZDogNSxcbiAgICAgICAgbGFiZWw6ICdMZXZlbCB0d28gMi0xJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIGlkOiA2LFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAyLTInLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICB7XG4gICAgaWQ6IDMsXG4gICAgbGFiZWw6ICdMZXZlbCBvbmUgMycsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgaWQ6IDcsXG4gICAgICAgIGxhYmVsOiAnTGV2ZWwgdHdvIDMtMScsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBpZDogOCxcbiAgICAgICAgbGFiZWw6ICdMZXZlbCB0d28gMy0yJyxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbl0pXG48L3NjcmlwdD5cblxuPHN0eWxlPlxuLmN1c3RvbS10cmVlLW5vZGUge1xuICBmbGV4OiAxO1xuICBkaXNwbGF5OiBmbGV4O1xuICBhbGlnbi1pdGVtczogY2VudGVyO1xuICBqdXN0aWZ5LWNvbnRlbnQ6IHNwYWNlLWJldHdlZW47XG4gIGZvbnQtc2l6ZTogMTRweDtcbiAgcGFkZGluZy1yaWdodDogOHB4O1xufVxuPC9zdHlsZT5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vcHJldmlldy0yMjA5Ni1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguY3NzJywgJ2h0dHBzOi8vcHJldmlldy0yMjA5Ni1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL3RoZW1lLWNoYWxrL2RhcmsvY3NzLXZhcnMuY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGxTZXR0bGVkKHN0eWxlcylcbn1cbiIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0yMjA5Ni1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0yMjA5Ni1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguY3NzIn19)

<img width="1279" height="441" alt="Image" src="https://github.com/user-attachments/assets/766b265d-4c0c-4645-8945-2085188ca2b1" />

[after](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiY3VzdG9tLXRyZWUtY29udGFpbmVyXCI+XG4gICAgPHA+VXNpbmcgcmVuZGVyLWNvbnRlbnQ8L3A+XG4gICAgPGVsLXRyZWVcbiAgICAgIHN0eWxlPVwibWF4LXdpZHRoOiA2MDBweFwiXG4gICAgICA6ZGF0YT1cImRhdGFTb3VyY2VcIlxuICAgICAgc2hvdy1jaGVja2JveFxuICAgICAgbm9kZS1rZXk9XCJpZFwiXG4gICAgICBkZWZhdWx0LWV4cGFuZC1hbGxcbiAgICAgIDpleHBhbmQtb24tY2xpY2stbm9kZT1cImZhbHNlXCJcbiAgICAgIDpyZW5kZXItY29udGVudD1cInJlbmRlckNvbnRlbnRcIlxuICAgIC8+XG4gICAgPHA+VXNpbmcgc2NvcGVkIHNsb3Q8L3A+XG4gICAgPGVsLXRyZWVcbiAgICAgIHN0eWxlPVwibWF4LXdpZHRoOiA2MDBweFwiXG4gICAgICA6ZGF0YT1cImRhdGFTb3VyY2VcIlxuICAgICAgc2hvdy1jaGVja2JveFxuICAgICAgbm9kZS1rZXk9XCJpZFwiXG4gICAgICBkZWZhdWx0LWV4cGFuZC1hbGxcbiAgICAgIDpleHBhbmQtb24tY2xpY2stbm9kZT1cImZhbHNlXCJcbiAgICA+XG4gICAgICA8dGVtcGxhdGUgI2RlZmF1bHQ9XCJ7IG5vZGUsIGRhdGEgfVwiPlxuICAgICAgICA8ZGl2IGNsYXNzPVwiY3VzdG9tLXRyZWUtbm9kZVwiPlxuICAgICAgICAgIDxzcGFuPnt7IG5vZGUubGFiZWwgfX08L3NwYW4+XG4gICAgICAgICAgPGRpdj5cbiAgICAgICAgICAgIDxlbC1idXR0b24gdHlwZT1cInByaW1hcnlcIiBsaW5rIEBjbGljaz1cImFwcGVuZChkYXRhKVwiPlxuICAgICAgICAgICAgICBBcHBlbmRcbiAgICAgICAgICAgIDwvZWwtYnV0dG9uPlxuICAgICAgICAgICAgPGVsLWJ1dHRvblxuICAgICAgICAgICAgICBzdHlsZT1cIm1hcmdpbi1sZWZ0OiA0cHhcIlxuICAgICAgICAgICAgICB0eXBlPVwiZGFuZ2VyXCJcbiAgICAgICAgICAgICAgbGlua1xuICAgICAgICAgICAgICBAY2xpY2s9XCJyZW1vdmUobm9kZSwgZGF0YSlcIlxuICAgICAgICAgICAgPlxuICAgICAgICAgICAgICBEZWxldGVcbiAgICAgICAgICAgIDwvZWwtYnV0dG9uPlxuICAgICAgICAgIDwvZGl2PlxuICAgICAgICA8L2Rpdj5cbiAgICAgIDwvdGVtcGxhdGU+XG4gICAgPC9lbC10cmVlPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBFbEJ1dHRvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuaW1wb3J0IHR5cGUgeyBSZW5kZXJDb250ZW50Q29udGV4dCwgUmVuZGVyQ29udGVudEZ1bmN0aW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5pbnRlcmZhY2UgVHJlZSB7XG4gIGlkOiBudW1iZXJcbiAgbGFiZWw6IHN0cmluZ1xuICBjaGlsZHJlbj86IFRyZWVbXVxufVxudHlwZSBOb2RlID0gUmVuZGVyQ29udGVudENvbnRleHRbJ25vZGUnXVxudHlwZSBEYXRhID0gUmVuZGVyQ29udGVudENvbnRleHRbJ2RhdGEnXVxuXG5sZXQgaWQgPSAxMDAwXG5cbmNvbnN0IGFwcGVuZCA9IChkYXRhOiBEYXRhKSA9PiB7XG4gIGNvbnN0IG5ld0NoaWxkID0geyBpZDogaWQrKywgbGFiZWw6ICd0ZXN0dGVzdCcsIGNoaWxkcmVuOiBbXSB9XG4gIGlmICghZGF0YS5jaGlsZHJlbikge1xuICAgIGRhdGEuY2hpbGRyZW4gPSBbXVxuICB9XG4gIGRhdGEuY2hpbGRyZW4ucHVzaChuZXdDaGlsZClcbiAgZGF0YVNvdXJjZS52YWx1ZSA9IFsuLi5kYXRhU291cmNlLnZhbHVlXVxufVxuXG5jb25zdCByZW1vdmUgPSAobm9kZTogTm9kZSwgZGF0YTogRGF0YSkgPT4ge1xuICBjb25zdCBwYXJlbnQgPSBub2RlLnBhcmVudFxuICBjb25zdCBjaGlsZHJlbjogVHJlZVtdID0gcGFyZW50Py5kYXRhLmNoaWxkcmVuIHx8IHBhcmVudD8uZGF0YVxuICBjb25zdCBpbmRleCA9IGNoaWxkcmVuLmZpbmRJbmRleCgoZCkgPT4gZC5pZCA9PT0gZGF0YS5pZClcbiAgY2hpbGRyZW4uc3BsaWNlKGluZGV4LCAxKVxuICBkYXRhU291cmNlLnZhbHVlID0gWy4uLmRhdGFTb3VyY2UudmFsdWVdXG59XG5cbmNvbnN0IHJlbmRlckNvbnRlbnQ6IFJlbmRlckNvbnRlbnRGdW5jdGlvbiA9IChoLCB7IG5vZGUsIGRhdGEgfSkgPT4ge1xuICByZXR1cm4gaChcbiAgICAnZGl2JyxcbiAgICB7XG4gICAgICBjbGFzczogJ2N1c3RvbS10cmVlLW5vZGUnLFxuICAgIH0sXG4gICAgW1xuICAgICAgaCgnc3BhbicsIG51bGwsIG5vZGUubGFiZWwpLFxuICAgICAgaCgnZGl2JywgbnVsbCwgW1xuICAgICAgICBoKFxuICAgICAgICAgIEVsQnV0dG9uLFxuICAgICAgICAgIHtcbiAgICAgICAgICAgIHR5cGU6ICdwcmltYXJ5JyxcbiAgICAgICAgICAgIGxpbms6IHRydWUsXG4gICAgICAgICAgICBvbkNsaWNrOiAoKSA9PiBhcHBlbmQoZGF0YSksXG4gICAgICAgICAgfSxcbiAgICAgICAgICB7IGRlZmF1bHQ6ICgpID0+ICdBcHBlbmQgJyB9XG4gICAgICAgICksXG4gICAgICAgIGgoXG4gICAgICAgICAgRWxCdXR0b24sXG4gICAgICAgICAge1xuICAgICAgICAgICAgdHlwZTogJ2RhbmdlcicsXG4gICAgICAgICAgICBsaW5rOiB0cnVlLFxuICAgICAgICAgICAgc3R5bGU6ICdtYXJnaW4tbGVmdDogNHB4JyxcbiAgICAgICAgICAgIG9uQ2xpY2s6ICgpID0+IHJlbW92ZShub2RlLCBkYXRhKSxcbiAgICAgICAgICB9LFxuICAgICAgICAgIHsgZGVmYXVsdDogKCkgPT4gJ0RlbGV0ZScgfVxuICAgICAgICApLFxuICAgICAgXSksXG4gICAgXVxuICApXG59XG5cbmNvbnN0IGRhdGFTb3VyY2UgPSByZWY8VHJlZVtdPihbXG4gIHtcbiAgICBpZDogMSxcbiAgICBsYWJlbDogJ0xldmVsIG9uZSAxJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBpZDogNCxcbiAgICAgICAgbGFiZWw6ICdMZXZlbCB0d28gMS0xJyxcbiAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICB7XG4gICAgICAgICAgICBpZDogOSxcbiAgICAgICAgICAgIGxhYmVsOiAnTGV2ZWwgdGhyZWUgMS0xLTEnLFxuICAgICAgICAgIH0sXG4gICAgICAgICAge1xuICAgICAgICAgICAgaWQ6IDEwLFxuICAgICAgICAgICAgbGFiZWw6ICdMZXZlbCB0aHJlZSAxLTEtMicsXG4gICAgICAgICAgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIGlkOiAyLFxuICAgIGxhYmVsOiAnTGV2ZWwgb25lIDInLFxuICAgIGNoaWxkcmVuOiBbXG4gICAgICB7XG4gICAgICAgIGlkOiA1LFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAyLTEnLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgaWQ6IDYsXG4gICAgICAgIGxhYmVsOiAnTGV2ZWwgdHdvIDItMicsXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG4gIHtcbiAgICBpZDogMyxcbiAgICBsYWJlbDogJ0xldmVsIG9uZSAzJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBpZDogNyxcbiAgICAgICAgbGFiZWw6ICdMZXZlbCB0d28gMy0xJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIGlkOiA4LFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAzLTInLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXSlcbjwvc2NyaXB0PlxuXG48c3R5bGU+XG4uY3VzdG9tLXRyZWUtbm9kZSB7XG4gIGZsZXg6IDE7XG4gIGRpc3BsYXk6IGZsZXg7XG4gIGFsaWduLWl0ZW1zOiBjZW50ZXI7XG4gIGp1c3RpZnktY29udGVudDogc3BhY2UtYmV0d2VlbjtcbiAgZm9udC1zaXplOiAxNHB4O1xuICBwYWRkaW5nLXJpZ2h0OiA4cHg7XG59XG48L3N0eWxlPlxuIiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9wcmV2aWV3LTIyMDk2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5jc3MnLCAnaHR0cHM6Ly9wcmV2aWV3LTIyMDk2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTIyMDk2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTIyMDk2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5jc3MifX0=)

tolking commented on packages/components/table/src/store/helper.ts:
```diff
@@ -12,7 +12,6 @@ const InitialStateMap = {
   selectOnIndeterminate: 'selectOnIndeterminate',
   indent: 'indent',
   lazy: 'lazy',
-  data: 'data',
```
I noticed that some places are using the `store.states.data`. won't this modification cause any problems?

[table.vue#L16-L17](https://github.com/element-plus/element-plus/blob/dev/packages/components/table/src/table.vue#L16-L17)

makedopamine committed:
Apply suggestion from @tolking

Co-authored-by: qiang <qw13131wang@gmail.com>

makedopamine committed:
docs: update example code

Co-authored-by: warmthsea <2586244885@qq.com>

tolking reviewed (approved):
Since the minimum Vue version has been updated, I think merging this PR in the next version would be more appropriate

warmthsea reviewed (approved):
LGTM

Dsaquel committed:
chore: remove csstype (#22487)

warmthsea committed:
Merge remote-tracking branch 'origin/dev' into dopamine/upgrade_vue

warmthsea committed:
chore: fix merge code type error

warmthsea committed:
chore: fix test:ssr error

- Cannot read properties of undefined (reading 'getSSRProps')

warmthsea committed:
chore: fix typecheck:vitest error

warmthsea committed:
chore: update pnpm yaml file

warmthsea committed:
test: fix collapse accordion error

warmthsea committed:
chore: update deps

warmthsea committed:
Merge branch 'dev' into dopamine/upgrade_vue

warmthsea committed:
chore: fix type error

btea committed:
Merge branch 'dev' into dopamine/upgrade_vue

xingyixiang commented:
https://github.com/element-plus/element-plus/blob/395af6672a15c5470b23a03020b8a6c669665c12/packages/components/tabs/src/tab-pane.vue#L28
Can `eagerComputed ` be replaced with `computed`


https://github.com/vueuse/vueuse/blob/8ff6f1948b68aab4e840de24678a38452d1e0c2d/packages/shared/computedEager/index.ts#L15

Dsaquel committed:
Merge branch 'dev' into dopamine/upgrade_vue

Dsaquel committed:
chore: lock file

Dsaquel committed:
chore: sync change

sync with the remove of vue macro

Dsaquel committed:
refactor: use computed instead of eagerComputed

Dsaquel committed:
fix: timeline.test.tsx typecheck

Dsaquel committed:
chore: clean lock file

try dont throw CodeFactor issues in ci

did:
- rm pnpm-lock.yaml
- rm -rf ./**/node_modules
- pnpm store prune
- pnpm cache delete
- pnpm install

Also stay in 3.1.0 for vue-tsc in order to avoid the warnings of
template refs, see https://github.com/vuejs/language-tools/issues/5815

Dsaquel committed:
chore: format code

Dsaquel reviewed (approved):
I think we're good although the CodeFactor report issues which i don't know if it's really a big deal 🤔 .
